### DISCUSSION TITLE
Is it possible to setup multi tenancy with copilot environments?

### DISCUSSION THREAD

**Discussion by joshgoodson** [aws/copilot-cli#2389] [Q&A] [answered]
Let's say I have several environments across multiple accounts.

Something like:

qa in an eng account
prod in a prod account (includes multi tenant services)
customer-env in a prod account (includes services that rely on multi tenant services)

is it possible to deploy services in the prod env that can be used or peered with a customer env in the same account?

**Comment by iamhopaul123** [comment, marked as answer]
Hello @joshgoodson since `prod` and `customer-env` are different environments, for now they must be in different VPCs. So they could only talk to each other using public internet (though the traffic can still be controlled by the LB security group). However, after https://github.com/aws/copilot-cli/pull/2394 hopefully we could address this issue by allowing environments to reuse the same VPC, which allows the services deployed within different envs can talk to each other using the service discovery.
